tket/src/Mapping/MappingFrontier.cpp

```diff
@@ -484,19 +483,21 @@ void MappingFrontier::update_linear_boundary_uids(
       // by type, label.first already assumed in circuit
       // this condition means label.second also in circuit
       // implies that a merging is done -> remove first qubit
-
-      if (this->linear_boundary->get<TagKey>().find(label.second) !=
-          this->linear_boundary->get<TagKey>().end()) {
-        // erase, assume updated already
-        this->linear_boundary->erase(label.first);
-      } else {
-        auto current_label_it =
-            this->linear_boundary->get<TagKey>().find(label.first);
-        // relabel "label.first" with "label.second"
-        this->linear_boundary->replace(
-            current_label_it, {label.second, current_label_it->second});
-        unit_map_t relabel = {label};
-        this->circuit_.rename_units(relabel);
+      if (this->reassignable_nodes_.find(Node(label.second)) ==
+          this->reassignable_nodes_.end()) {
```

Comment:
I don't know if it can? The second `if` is followed by an `else` statement. Could you show me the logic please?

Reply:
Yes, I mean this
```
if (label.first != label.second && this->reassignable_nodes_.find(Node(label.second)) == this->reassignable_nodes_.end()) {
          if(...){
          }
          else {
          }
 }
```

Reply:
We don't want to do the `else` if `label.first !=label.second` isn't true so I'm not sure this helps? Sorry I'm probably missing something really obvious here if you can attempt to explain it in even simpler terms

Reply:
Sorry, I think we are probably talking about two different if statements.

You have this logic:
```
//first if
if (label.first != label.second){
  //second if
  if (this->reassignable_nodes_.find(Node(label.second)) == this->reassignable_nodes_.end()) {
    // here goes the the code with the if else structure.
  }
}
```
What I meant was that since the `second if` doesn't have a else statement you can combine the `first if` and the `second if`:
```
//first if and second if
if (label.first != label.second && this->reassignable_nodes_.find(Node(label.second)) == this->reassignable_nodes_.end()){
    // here goes the the code with the if else structure.
}
```
The actual code will become:
```
// implies new labelling
if (label.first != label.second &&
    this->reassignable_nodes_.find(Node(label.second)) ==
        this->reassignable_nodes_.end()) {
  // by type, label.first already assumed in circuit
  // this condition means label.second also in circuit
  // implies that a merging is done -> remove first qubit
  if (this->linear_boundary->get<TagKey>().find(label.second) !=
      this->linear_boundary->get<TagKey>().end()) {
    // erase, assume updated already
    this->linear_boundary->erase(label.first);
  } else {
    auto current_label_it =
        this->linear_boundary->get<TagKey>().find(label.first);
    // relabel "label.first" with "label.second"
    this->linear_boundary->replace(
        current_label_it, {label.second, current_label_it->second});
    unit_map_t relabel = {label};
    this->circuit_.rename_units(relabel);
  }
}
```
This really doesn't matter at all. I just thought there might be an unnecessary nesting.

Reply:
This is embarrassing - I was looking at the wrong branch locally....
Yes everything you're saying makes sense, I will change it!
Sorry for tomfoolery, oopsiess